### PR TITLE
perl 5.36.0

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Starting over on #102549 since some changes have gone in other PRs like #108663.

Planning to cherry-pick the commits from other PR once we have readout of linkage failures.

### CODE REVIEW

**Comment by danielnachun** [comment]
Perl 5.37 is out now, so we should try that instead.

**Comment by danielnachun** [comment]
Also from prior testing, the self-hosted runner is not needed just for dependent testing, so you might be able to run this faster if you remove that flag.

**Comment by cho-m** [comment]
> Perl 5.37 is out now, so we should try that instead.

Odd numbers are development versions. Latest stable is still 5.36.

File: Formula/perl.rb [hunk 1]
```diff
@@ -54,13 +53,10 @@ def install
       -Duselargefiles
       -Dusethreads
     ]
```
Comment (cho-m) [inline Formula/perl.rb:55]: Can probably add `-Dinstallstyle=lib/perl5` as alternative fix for #112150

**Comment by danielnachun** [comment]
> > Perl 5.37 is out now, so we should try that instead.
> 
> Odd numbers are development versions. Latest stable is still 5.36.

Ah I did not know that.  Should we add a comment explaining that?

File: Formula/perl.rb (hunk 1 above)
Comment (danielnachun) [reply Formula/perl.rb:55]: I was going to suggest in that PR that we see if we can avoid patching to change the `installstyle` and it appears we can.  I think this is a better approach and should be used instead.

File: Formula/biber.rb [hunk 2]
```diff
@@ -540,30 +542,27 @@ def install
...
-
-    bin.install Dir[libexec/"bin/*"]
-    bin.env_script_all_files(libexec/"bin", PERL5LIB: ENV["PERL5LIB"])
+    bin_after = Dir[libexec/"bin/*"].to_set
+    (bin_after - bin_before).each do |file|
+      basename = Pathname(file).basename
+      (bin/basename).write_env_script file, PERL5LIB: ENV["PERL5LIB"]
+    end
     man1.install libexec/"man/man1/biber.1"
+    (pkgshare/"test").install "t/tdata/annotations.bcf", "t/tdata/annotations.bib"
```
Comment (cho-m) [inline Formula/biber.rb:555]: Just picked the first test from directory. Not sure if there is a better one. We've been hit a few times by upstream modifying the `test.bcf` file with new releases. Unless we save a copy, it didn't seem worth seeing CI failures related to checksum mismatches or version incompatibility.

Comment (danielnachun) [reply Formula/biber.rb:555]: I didn't realize there was test directory but that seems like a much more reliable solution.

**Comment by danielnachun** [comment]
Linux-specific failures:
- subversion: top-level mandir
- feedgnuplot: top-level mandir
- grokj2k: we're on a very old release - the newest release builds fine on Linux
- pidgin: needs a fix for `XML::Parser` - it should use `ENV.prepend_path` instead of `ENV.prepend` when adding the library path of `intltool` to `PERL5LIB`.

**Comment by cho-m** [comment]
> * grokj2k: we're on a very old release - the newest release builds fine on Linux

I think we've had issues on macOS when trying to update `grokj2k`. Right after I fixed formula in #95683 (`9.2.0 --> 9.7.1`), upstream broke build in next patch #96168 and we haven't been able to update since.

Not sure about current version, but it was one of failures in previous `perl` PR #102549 so I want to avoid blocking `perl` update on this.

---

> * pidgin: needs a fix for `XML::Parser` - it should use `ENV.prepend_path` instead of `ENV.prepend` when adding the library path of `intltool` to `PERL5LIB`.

I was previously wondering if we should enable `perl` vendor directory and create a `perl-xml-parser` formula just to avoid all these workarounds due to common `intltool.m4` script.

Though, most formulae are moving away from `intltool` so probably not important.

**Comment by cho-m** [comment]
Looks like `abricate` is failing everywhere. Given we don't have a Monterey bottle, it was probably broken since. I might try switching to macOS Perl as the Perl dependencies are pre-installed. Relying on `cpanminus` may have made the build harder to reproduce.

`man-db` failed in previous PR too. Will need to determine if it needs a revision bump. Test failure output isn't very useful.

**Comment by cho-m** [comment]
`pidgin` is actually due to `intltool` failure (XML::Parser is a compiled library)
```console
$ PERL5LIB=/home/linuxbrew/.linuxbrew/opt/intltool/libexec/lib/perl5 perl -e "require XML::Parser"
Expat.c: loadable library and perl binaries are mismatched (got first handshake key 0xeb00080, needed 0xeb80080)

$ file /home/linuxbrew/.linuxbrew/opt/intltool/libexec/lib/perl5/x86_64-linux-thread-multi/auto/XML/Parser/Expat/Expat.so
/home/linuxbrew/.linuxbrew/opt/intltool/libexec/lib/perl5/x86_64-linux-thread-multi/auto/XML/Parser/Expat/Expat.so: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=ad3c15e5bc2cbc090fb4422af4e1e72ab735677e, not stripped
```

File: Formula/subversion.rb [hunk 3]
```diff
@@ -223,8 +225,12 @@ def install
...
       end
     end
-    system "make", "swig-pl"
-    system "make", "install-swig-pl"
+    system "make", "swig-pl-lib"
+    system "make", "install-swig-pl-lib"
+    cd "subversion/bindings/swig/perl/native" do
+      system perl, "Makefile.PL", "PREFIX=#{prefix}", "INSTALLSITEMAN3DIR=#{man3}"
+      system "make", "install"
+    end
```
Comment (cho-m) [inline Formula/subversion.rb:233]: Alternatively could have
- manually moved files (`man3.install ...`) or
- modified the Makefile
    ```rb
    inreplace "Makefile", "$(PERL) Makefile.PL PREFIX=$(prefix)", "\\0 INSTALLSITEMAN3DIR=#{man3}"
    ```

The current approach is officially documented method for adding options during Perl build https://github.com/apache/subversion/blob/1.14.2/subversion/bindings/swig/INSTALL#L217-L236

`PREFIX` is from the Makefile https://github.com/apache/subversion/blob/1.14.2/Makefile.in#L883-L884 while `INSTALLSITEMAN3DIR` is to fix the manpage installation.

File: Formula/grokj2k.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@ class Grokj2k < Formula
   url "https://github.com/GrokImageCompression/grok/archive/v9.7.1.tar.gz"
   sha256 "a7d433dca92b035349ef8203eb44cb6d0b2c9b41aecd2d12872a9ca2761e0606"
   license "AGPL-3.0-or-later"
-  revision 1
+  revision 2
```
Comment (cho-m) [inline Formula/grokj2k.rb:8]: Hopefully, only remaining failure will be Linux `grokj2k`. I don't think we can fix in this PR so may need to follow up afterward.

Comment (danielnachun) [reply Formula/grokj2k.rb:8]: Agreed, we should follow up with this in a PR to upgrade `grokj2k`.

File: Formula/subversion.rb (hunk 3 above)
Comment (danielnachun) [reply Formula/subversion.rb:233]: The documented approach looks pretty clean so it seems fine to me.

File: Formula/grokj2k.rb (hunk 4 above)
Comment (cho-m) [reply Formula/grokj2k.rb:8]: Working on it in #112161. Current problem is macOS Perl headers conflict with C++ headers. It builds for me if I disable `exiftool` feature. Need to look into it a bit more:
```
/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/c++/v1/locale:3569:35: error: too few arguments provided to function-like macro invocation
        return do_open(__nm, __loc);
                                  ^
/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/System/Library/Perl/5.30/darwin-thread-multi-2level/CORE/perl.h:6924:9: note: macro 'do_open' defined here
#define do_open(g, n, l, a, rm, rp, sf) \
        ^
```

**Comment by cho-m** [comment]
I think this is ready to merge once CI finishes. macOS is all passing while Linux currently only failed on `grokj2k`, which should be fixed in #112269.

**Review by danielnachun — DISMISSED** [review]
Everything is passing on macOS and as hoped the only Linux failure is `grokj2k`.  Let get this merged.

**Pull request closed without merging**